packages/@react-stately/table/src/useTreeGridState.ts

```diff
@@ -10,73 +10,91 @@
  * governing permissions and limitations under the License.
  */
 
+import {CollectionBuilder} from '@react-stately/collections';
 import {Expandable} from '@react-types/shared';
-import {TableCollection as ITableCollection} from '@react-types/table';
-import {ITreeGridCollection, TreeGridCollection} from './TreeGridCollection';
-import {Key, useCallback, useMemo} from 'react';
+import {GridNode} from '@react-types/grid';
+import {Key, useMemo} from 'react';
+import {TableCollection} from './TableCollection';
 import {TableState, TableStateProps, useTableState} from './useTableState';
-import {useCollection} from '@react-stately/collections';
 import {useControlledState} from '@react-stately/utils';
 
 export interface TreeGridState<T> extends TableState<T> {
   /** A set of keys for items that are expanded. */
   expandedKeys: 'all' | Set<Key>,
   /** Toggles the expanded state for a row by its key. */
   toggleKey(key: Key): void,
-  collection: ITreeGridCollection<T>
+  /** The key map containing nodes representing the collection's tree grid structure. */
+  keyMap: Map<Key, GridNode<T>>,
+  /** The number of columns provided by the user. */
+  userColumnCount: number
 }
 
-export interface TreeGridStateProps<T> extends Expandable, TableStateProps<T> {
-  collection?: ITreeGridCollection<T>
-}
+// TODO: For now I am omitting collection since this is a bit of a special case where are still using TableCollection but include
+// extra information for the tree grid in the state. That means the user can't really provide their own collection. Perhaps it would
+// be good to still have a TreeGridCollection implementation in that case?
+export interface TreeGridStateProps<T> extends Expandable, Omit<TableStateProps<T>, 'collection'> {}
```

Comment:
Since we are generating a key map to represent the tree grid structure and then flattening the rows for TableCollection instead of making a TreeGridCollection, it doesn't really make sense IMO to accept a user generated collection here (or at the very least it doesn't feel very straight forward to accept one). If a user were to provide a collection that we would use over generating a separate TableCollection, it would have to contain the same information as the TableCollection with flattened rows for layout/keyboardDelegate purpose and provide the extra tree grid key map info. Perhaps an argument for keeping TreeGridCollection and the extra changes to layout/keyboard delegates to accommodate the key getters that exist in the original PR